crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -392,6 +414,40 @@ impl<'a> Visitor<'a> {
         Ok(errors)
     }
 
+    /// Finishes visiting the tasks, creates the run summary, and either
+    /// prints, saves, or sends it to spaces.
+    pub(crate) async fn finish(self) -> Result<(), Error> {
```

Comment:
I think we want to take `exit_code` as a parameter here as the `visit` method returns and exit code and we can plumb it in.